packages/react/src/I18nProvider.tsx

```diff
@@ -1,15 +1,15 @@
 import React, { FunctionComponent } from "react"
-import { I18n } from "@lingui/core"
-import { TransRenderType } from "./Trans"
+import type { I18n } from "@lingui/core"
+import type { TransRenderType } from "./Trans"
 
 interface I18nContext {
   i18n: I18n
   defaultRender?: TransRenderType
 }
 
-export interface I18nProviderProps extends I18nContext {}
+export type I18nProviderProps = I18nContext
```

Comment:
Just stumbled upon a perfect article for you today if you think you cannot extend type :)

https://ultimatecourses.com/blog/typescript-interfaces-vs-types

Ultimately, in most cases, it does boil down to personal preference. For me it's based on the fact that `type` can do pretty much everything (and more) compared to `interface`. Besides there is also `import type` since 3.8, no `import interface`. Using two different keywords becomes opinionated battle when use which. So I opted to `type` everywhere and it's a win.